docs/BMI-Installation.md

```diff
@@ -0,0 +1,283 @@
+# BMI
+
+The Bare Metal Imaging (BMI) is a core component of the Massachusetts Open Cloud and a image management system(IMS) that  
+* provisions numerous nodes as quickly as possible while preserving support for multitenancy using Hardware as a Service (HIL) and  
+* introduces the image management techniques that are supported by virtual machines, with little to no impact on application performance.  
+
+***
+## Installation
+
+This document describes how to setup BMI
+
+### Prerequisite Softwares
+
+BMI requires the following non python dependencies. 
+
+#### Ceph Client
+Ceph is a distributed object store and file system designed for performance and scalibility. 
+For more information, visit the [official website](http://docs.ceph.com/docs/master/start/)
+
+You can follow [this guide](http://palmerville.github.io/2016/04/30/single-node-ceph-install.html) 
+to setup a ceph client on a VM
+
+#### iSCSI Server
+We recommend using the TGT iSCSI variant as its driver is more stable.   
+You can install it with the following set of commands:  
+
+```
+git clone https://github.com/fujita/tgt.git
+cd tgt
+
+yum install -y make
+yum install -y gcc
+** delete install doc from Makefile if problem occurs in the next three steps
+make CEPH_RBD=1 clean
+make CEPH_RBD=1
+make CEPH_RBD=1 install
+
+
+cp scripts/tgtd.service /usr/lib/systemd/system/
+systemctl daemon-reload
+systemctl list-unit-files | grep -i tgt
+
+iptables -I INPUT -p tcp -m tcp --dport 3260 -j ACCEPT
+service iptables save
+
+systemctl start tgtd
+chkconfig tgtd on
+systemctl status tgtd
+```
+
+For more information visit their [website](http://stgt.sourceforge.net/) and the [Quick Start Guide](https://fedoraproject.org/wiki/Scsi-target-utils_Quickstart_Guide)
+
+#### DHCP server
+You can setup `dnsmasq` for this. 
+
+* On ubuntu:
+`$ sudo apt-get install dnsmasq`
+
+* On CentOS:
+`$ yum install dnsmasq`
+
+For more information about dnsmasq, you can look at the [wiki](https://wiki.debian.org/HowTo/dnsmasq)
+
+#### TFTP Server
+##### On ubuntu:
+* Install these packages:  `$ sudo apt-get install xinetd tftpd tftp`
```

Comment:
Why do we need a explicit TFTP, dnsmasq is going to bring that in. Unless, we go for another dnsserver, this section should be optional.